backend/tournesol/tests/test_api_unconnected_entities.py

```diff
@@ -0,0 +1,248 @@
+            entity_1=video_1,
+            entity_2=video_3,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_2,
+            entity_2=video_3,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_4,
+            entity_2=video_5,
+        )
+
+    def test_should_return_non_connected_entity(self):
+        self.client.force_authenticate(self.user_1)
+
+        response = self.client.get(
+            "{}/{}/".format(self.user_base_url, self.video_source.id),
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2)
+        self.assertEqual(
+                list(map(lambda x: x["uid"], response.data["results"])),
+                list(map(lambda x: x.uid, self.unrelated_video))
+        )
+
+
+class AdvancedNotAllConnectedTest(TestCase):
+    """
+    TestCase for the unconnected entities API.
+    """
+    def setUp(self):
+        self.client = APIClient()
+        self.user_1 = UserFactory()
+        user_2 = UserFactory()
+        self.poll_videos = Poll.default_poll()
+        self.user_base_url = "/users/me/unconnected_entities/{}".format(
+            self.poll_videos.name
+        )
+
+        video_1 = VideoFactory()
+        video_2 = VideoFactory()
+        video_3 = VideoFactory()
+        video_4 = VideoFactory()
+        video_5 = VideoFactory()
+        video_6 = VideoFactory()
+        video_7 = VideoFactory()
+
+        self.unrelated_video = [video_6, video_7]
+
+        self.video_source = video_1
+
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_4,
+            entity_2=video_5,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_2,
+            entity_2=video_4,
+        )
+        ComparisonFactory(
+            user=user_2,
+            entity_1=video_2,
+            entity_2=video_6,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_7,
+            entity_2=video_6,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_3,
+            entity_2=video_2,
+        )
+        ComparisonFactory(
+            user=self.user_1,
+            entity_1=video_3,
+            entity_2=video_1,
+        )
+
+    def test_should_return_non_connected_entity(self):
+        self.client.force_authenticate(self.user_1)
+
+        response = self.client.get(
+            "{}/{}/".format(self.user_base_url, self.video_source.id),
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2)
+        self.assertEqual(
+                list(map(lambda x: x["uid"], response.data["results"])),
+                list(map(lambda x: x.uid, self.unrelated_video))
```

Comment:
Yes, I also recommend using the full word `entity` if the resulting statement fits a single line.

- `ent` is acceptable I think if it makes the statement fit on a single line
- `e` is not pythonic
- `x` is confusing

In case of doubt, we can always refer to the Zen of Python (and PEP8): https://peps.python.org/pep-0020/

Reply:
with entity, the line is 82, which is black compliant and acceptable if it's for readability